### DISCUSSION TITLE
Question:Do PQ-KEMs kyber and KEMs approved for 4th round , need Domain Separation for deriving multiple keys from shared secret?

### DISCUSSION THREAD

**Discussion by evil-user** [open-quantum-safe/liboqs#1262] [Q&A] [answered]
Do PQ-KEMs kyber and KEMs approved for 4th round by NIST, need Domain Separation for deriving multiple keys from shared secret? I am using HKDF to split shared secret to derive multiple keys, one for encryption , the other one for HMAC.

**Comment by dstebila** [comment, marked as answer]
Any cryptographic key, including those output by KEMs, should only be used for a single purpose. If you need to use multiple symmetric algorithms (e.g., AES for encryption and then HMAC for authentication), you should use a key derivation function to derive multiple computationally independent keys from the single key.

**Reply by evil-user** [reply]
@dstebila a small doubt does deriving key for AES and deriving separate key for HMAC using HKDF on shared secret obtained from PQ-KEM need domain separation?

**Reply by dstebila** [reply]
KDFs take labels as input which you should use for domain separation. This is generally outside the scope of liboqs so if you have more questions about how to use KDFs I suggest trying StackExchange.
